python/dolfinx/mesh.py

```diff
@@ -65,10 +64,141 @@
 ]
 
 
+class Topology:
+    """Topology for a :class:`dolfinx.mesh.Mesh`"""
+
+    _cpp_object: _cpp.mesh.Topology
+
+    def __init__(self, topology: _cpp.mesh.Topology):
+        """Initialize a topology from a C++ topology.
+        Args: The C++ topology object
+        Note:
+            topology objects should not usually be created using this
```

Comment:
Also leave an explicit pointer to the constructor here.

Reply:
We only have a C++ binding for a constructor atm, so leaving that here for now.